R-package/tests/testthat/test_basic.R

```diff
@@ -1021,3 +1021,73 @@ test_that("using lightgbm() without early stopping, best_iter and best_score com
   expect_identical(bst$best_iter, which.max(auc_scores))
   expect_identical(bst$best_score, auc_scores[which.max(auc_scores)])
 })
+
+test_that("lgb.train() throws an informative error if interaction_constraints is not a list", {
+  dtrain <- lgb.Dataset(train$data, label = train$label)
+  params <- list(objective = "regression", interaction_constraints="[1,2],[3]")
+    expect_error({
+      bst <- lightgbm(
+        data = dtrain
+        , params = params
+        , nrounds = 2
+      )
+    }, "interaction_constraints must be a list")
+})
+
+test_that("lgb.train() throws an informative error if the members of interaction_constraints are not character or numeric vectors", {
+  dtrain <- lgb.Dataset(train$data, label = train$label)
+  params <- list(objective = "regression", interaction_constraints=list(list(1L, 2L), list(3L)))
+    expect_error({
+      bst <- lightgbm(
+        data = dtrain
+        , params = params
+        , nrounds = 2
+      )
+    }, "every element in interaction_constraints must be a character vector or numeric vector")
+})
+
+test_that("lgb.train() throws an informative error if interaction_constraints contains a too large index", {
+  dtrain <- lgb.Dataset(train$data, label = train$label)
+  params <- list(objective = "regression",
+                 interaction_constraints=list(list(1L, length(colnames(train$data)) + 1), list(3L)))
+    expect_error({
+      bst <- lightgbm(
+        data = dtrain
+        , params = params
+        , nrounds = 2
+      )
+    }, "every element in interaction_constraints must be a character vector or numeric vector")
+})
```

Comment:
I don't think this test is right. We shouldn't get the "must be a character or numeric vector" for this one, but an error like `supplied a too large value in interaction_constraints`

Reply:
Fixed, thanks. It was actually passing before despite this mistake, but that's because I had another error in the test!